### PR TITLE
feat(BA-3954): Implement bulk_purge_users mutation (Strawberry GQL)

### PR DESCRIPTION
**Pull request by MinJuTur**
## Summary
- Implement `admin_bulk_purge_users_v2` Strawberry GraphQL mutation for bulk user permanent deletion
- Add full-stack implementation across action, service, repository, processor, and GQL resolver layers
- Support partial failure handling: successfully purged users are reported alongside individual failures
- Include `purge_shared_vfolders` and `delegate_endpoint_ownership` global options
- Add unit tests covering all success, partial failure, all failure, and empty list scenarios

## Related Issue
- Jira: BA-3954

## Test plan
- [x] Unit tests for `UserService.bulk_purge_users` (all success, partial failure, all failure, empty list)
- [x] `pants lint` passed
- [x] `pants check` (mypy) passed
- [x] `pants test` passed

🤖 Generated with [Claude Code](https://claude.com/claude-code)

----
📚 Documentation preview 📚: https://sorna--9159.org.readthedocs.build/en/9159/

----
📚 Documentation preview 📚: https://sorna-ko--9159.org.readthedocs.build/ko/9159/

### CODE REVIEW

File: src/ai/backend/manager/services/user/service.py [hunk 1]
```diff
@@ -195,6 +199,98 @@ async def purge_user(self, action: PurgeUserAction) -> PurgeUserActionResult:
 
         return PurgeUserActionResult()
 
+    async def _purge_single_user(
```
Comment (MinJuTur) [inline src/ai/backend/manager/services/user/service.py:213]: The existing `purge_user()` is email-based, while the new `_purge_single_user()` is UUID-based. The rest of the purge logic is identical.

File: src/ai/backend/manager/api/gql/user/resolver/mutation.py [hunk 2]
```diff
@@ -282,25 +289,62 @@ async def admin_purge_user(
...
     """
-    raise NotImplementedError("admin_purge_users is not yet implemented")
+    check_admin_only()
+    ctx = info.context
+
+    me = current_user()
+    if me is None or ctx.db is None:
+        raise RuntimeError("User context or database not available")
+    user_repo = UserRepository(db=ctx.db)
+    admin_user = await user_repo.get_user_by_uuid(me.user_id)
```
Comment (MinJuTur) [inline src/ai/backend/manager/api/gql/user/resolver/mutation.py]: `current_user()` only provides `user_id`, so we query `UserRepository.get_user_by_uuid()` to retrieve the full admin user info (`uuid`,` email`, `main_access_key`) required by `UserInfoContext`.

File: src/ai/backend/manager/api/gql/user/resolver/mutation.py [hunk 3]
```diff
@@ -282,25 +289,62 @@ async def admin_purge_user(
...
-        NotImplementedError: This mutation is not yet implemented.
+        BulkPurgeUsersV2PayloadGQL with purged count and failures.
     """
-    raise NotImplementedError("admin_purge_users is not yet implemented")
+    check_admin_only()
+    ctx = info.context
+
+    me = current_user()
+    if me is None or ctx.db is None:
+        raise RuntimeError("User context or database not available")
```
Comment (jopemachine) [inline src/ai/backend/manager/api/gql/user/resolver/mutation.py]: Raise BackendAIError-based exception

File: src/ai/backend/manager/services/user/service.py [hunk 4]
```diff
@@ -195,6 +199,98 @@ async def purge_user(self, action: PurgeUserAction) -> PurgeUserActionResult:
...
+    ) -> BulkPurgeUserActionResult:
+        purged_user_ids: list[UUID] = []
+        failures: list[BulkPurgeError] = []
+
+        for user_uuid in action.user_ids:
+            try:
+                await self._purge_single_user(user_uuid, action)
+                purged_user_ids.append(user_uuid)
+            except Exception as e:
+                log.warning(f"Failed to purge user {user_uuid}: {e}")
```
Comment (jopemachine) [inline src/ai/backend/manager/services/user/service.py]: Why not log.error?

File: src/ai/backend/manager/api/gql/user/types/payloads.py [hunk 5]
```diff
@@ -113,3 +113,27 @@ class PurgeUsersPayloadGQL:
     failed_user_ids: list[UUID] = strawberry.field(
         description="List of user UUIDs that failed to purge, if any."
     )
+
+
+@strawberry.type(
+    name="BulkPurgeUserError",
```
Comment (jopemachine) [inline src/ai/backend/manager/api/gql/user/types/payloads.py]: BulkPurgeUserV2Error

**Comment by jopemachine** [comment]
Seems lots of conflict occurred

File: src/ai/backend/manager/api/gql/user/resolver/mutation.py (hunk 3 above)
Comment (MinJuTur) [reply src/ai/backend/manager/api/gql/user/resolver/mutation.py]: > Raise BackendAIError-based exception

This check is unreachable at runtime since `check_admin_only()` already raises `HTTPForbidden` if me is None. It's only here for mypy type narrowing. Which BackendAIError subclass would you prefer?

File: src/ai/backend/manager/api/gql/user/resolver/mutation.py [hunk 6]
```diff
@@ -282,25 +289,62 @@ async def admin_purge_user(
...
+    user_repo = UserRepository(db=ctx.db)
```
Comment (jopemachine) [inline src/ai/backend/manager/api/gql/user/resolver/mutation.py]: Why UserRepository is being created at mutation?

File: src/ai/backend/manager/api/gql/user/resolver/mutation.py (hunk 3 above)
Comment (jopemachine) [reply src/ai/backend/manager/api/gql/user/resolver/mutation.py]: Let’s add a dedicated exception that clearly indicates the code path is unreachable, and raise it there.

File: src/ai/backend/manager/api/gql/user/resolver/mutation.py [hunk 7]
```diff
@@ -282,25 +289,62 @@ async def admin_purge_user(
...
-    Raises:
-        NotImplementedError: This mutation is not yet implemented.
+        BulkPurgeUsersV2PayloadGQL with purged count and failures.
     """
-    raise NotImplementedError("admin_purge_users is not yet implemented")
+    check_admin_only()
+    ctx = info.context
+
+    me = current_user()
+    if me is None or ctx.db is None:
```
Comment (jopemachine) [inline src/ai/backend/manager/api/gql/user/resolver/mutation.py]: I’m not sure why db is included in the context (ctx).
Please don’t use it.

File: docs/manager/graphql-reference/v2-schema.graphql [hunk 8]
```diff
@@ -930,6 +930,40 @@ type BulkCreateUsersV2Payload {
...
+  """List of user UUIDs to purge."""
+  userIds: [UUID!]!
+
+  """
+  If true, migrate shared virtual folders to the admin user before purging.
+  """
+  purgeSharedVfolders: Boolean! = false
+
+  """If true, delegate endpoint ownership to the admin user before purging."""
+  delegateEndpointOwnership: Boolean! = false
```
Comment (HyeockJinKim) [inline docs/manager/graphql-reference/v2-schema.graphql:961]: I'd like to bundle these options—what do you think?

Comment (jopemachine) [reply docs/manager/graphql-reference/v2-schema.graphql:961]: Please create `BulkPurgeUsersV2Options` type

**Review by jopemachine — CHANGES_REQUESTED** [review]
